### ISSUE TITLE
[Discussion] Update defaults

### ISSUE THREAD

**Issue by saccarosium** [lervag/wiki.vim#287]
related: https://github.com/lervag/wiki-ft.vim/issues/14

As mention in the related issue made on the `wiki-ft` repo, we defaults to `.wiki` file extension but the syntax that we check in functions such as `wiki#toc#gather_entries` for markdown syntax.

So my proposal for updating the defaults are:
1. making markdown the default filetype and the extension `.md` the default;
2. make the default tag parser one who checks for `#tag`;
3. refactoring vim script APIs for support other filetypes.

**Comment by lervag** [comment]
> So my proposal for updating the defaults are:
>
> 1. making markdown the default filetype and the extension `.md` the default;

Yes, I think you are right. This seems sensible.

> 2. make the default tag parser one who checks for `#tag`;

Is `#tag` a more standard convention for tags in Markdown? I don't know if I've seen any real specifications for tags, to be honest. Do you have any references?

`#...` does sort of "clash" with the setext header. Thus the `:tag:` syntax seems equally good to me.

> 3. refactoring vim script APIs for support other filetypes.

This is unclear. It would be helpful if you could be more specific here. Note, I'm not saying I don't agree, I just want help to 1) understand how you think I should refactor and 2) which APIs/features.

**Comment by saccarosium** [comment]
> Is #tag a more standard convention for tags in Markdown? I don't know if I've seen any real specifications for tags, to be honest. Do you have any references?

It isn't specified in the specification, but, basically every application that uses markdown and support tags uses that syntax. But this is debatable.

> This is unclear. It would be helpful if you could be more specific here. Note, I'm not saying I don't agree, I just want help to 1) understand how you think I should refactor and 2) which APIs/features.

One that comes to mind is `toc#gather_entries` that even do I use asciidoc as my primary filetype searches of atx stile headings and when I use `WikiToc` doesn't show anything.

**Comment by lervag** [comment]
> > Is #tag a more standard convention for tags in Markdown? I don't know if I've seen any real specifications for tags, to be honest. Do you have any references?
> 
> It isn't specified in the specification, but, basically every application that uses markdown and support tags uses that syntax. But this is debatable.

Yes, it _is_ debatable. I've tried to find specific examples of any kind of tag support. I've found some seemingly relevant links:
* https://stackoverflow.com/questions/8783641/semantic-tagging-in-markdown
* [Vimwiki docs about tags](https://github.com/vimwiki/vimwiki/blob/dev/doc/vimwiki.txt#L1568) (Vimwiki also supports Markdown, but it does not allow any other tag formats.)

I'm not sure I really want to change this. I've already generalized the concept to allow custom tag definitions, so if you want to use the syntax `#tag`, it should not be hard to allow this with user customization.

> One that comes to mind is `toc#gather_entries` that even do I use asciidoc as my primary filetype searches of atx stile headings and when I use `WikiToc` doesn't show anything.

Yes; `toc#gather_entries` is hard coded for Markdown right now. So that's a good example. Let me know if you find other examples.

**Comment by lervag** [comment]
I've created two issues to follow this up now.

**Comment by saccarosium** [comment]
> Yes, it is debatable. I've tried to find specific examples of any kind of tag support. I've found some seemingly relevant links:

Fair enough I will drop this request... You won XD.

**Comment by lervag** [comment]
Heh; I don't mind loosing, but I do want to be properly convinced first :)

Can we close this discussion now? Let me know if you notice other things that feel wrong wrt. the "not a filetype plugin" concept!
